### DISCUSSION TITLE
Matching a fan curve in Openmodelica

### DISCUSSION THREAD

**Discussion by DynteqMS** [lbl-srg/modelica-buildings#3370] [Q&A] [answered]
<html>
<body>
<p style="margin: 0px 0px 1.1em; padding: 0px; border: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-alternates: inherit; font-weight: 400; font-stretch: inherit; line-height: inherit; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI Adjusted&quot;, &quot;Segoe UI&quot;, &quot;Liberation Sans&quot;, sans-serif; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 15px; vertical-align: baseline; box-sizing: inherit; clear: both; --_pr-img-mb:0; color: rgb(35, 38, 41); letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Trying to optimize some ducting configuration I want to model a system of ducts with a fan to simulate the flow and pressure distribution.<br style="box-sizing: inherit;">In order to choose the right fan I want to add a fan curve from the fan supplier. But how to get the fan to behave like the curve says it should?</p><p style="margin: 0px 0px 1.1em; padding: 0px; border: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-alternates: inherit; font-weight: 400; font-stretch: inherit; line-height: inherit; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI Adjusted&quot;, &quot;Segoe UI&quot;, &quot;Liberation Sans&quot;, sans-serif; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 15px; vertical-align: baseline; box-sizing: inherit; clear: both; --_pr-img-mb:0; color: rgb(35, 38, 41); letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Using a SpeedControlled_y pump from the Buildings library I want to match this performance curve:</p><p style="margin: 0px 0px 1.1em; padding: 0px; border: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-alternates: inherit; font-weight: 400; font-stretch: inherit; line-height: inherit; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI Adjusted&quot;, &quot;Segoe UI&quot;, &quot;Liberation Sans&quot;, sans-serif; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 15px; vertical-align: baseline; box-sizing: inherit; clear: both; --_pr-img-mb:0; color: rgb(35, 38, 41); letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a href="https://i.stack.imgur.com/GU4pr.png" rel="nofollow noreferrer" style="margin: 0px; padding: 0px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 15px; vertical-align: baseline; --_li-fc:var(--theme-link-color); --_li-fc-hover:var(--theme-link-color-hover); --_li-fc-visited:var(--theme-link-color-visited); color: var(--theme-link-color); cursor: pointer; text-decoration: underline; user-select: auto; box-sizing: inherit;"><img src="https://i.stack.imgur.com/GU4pr.png" alt="enter image description here" style="margin: 0px; padding: 0px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 15px; vertical-align: bottom; box-sizing: inherit; max-width: 100%;"></a></p><p style="margin: 0px 0px 1.1em; padding: 0px; border: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-alternates: inherit; font-weight: 400; font-stretch: inherit; line-height: inherit; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI Adjusted&quot;, &quot;Segoe UI&quot;, &quot;Liberation Sans&quot;, sans-serif; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 15px; vertical-align: baseline; box-sizing: inherit; clear: both; --_pr-img-mb:0; color: rgb(35, 38, 41); letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">The test setup was adapted from Buildings.Fluid.Movers.Examples.ClosedLoop_y<br style="box-sizing: inherit;"><a href="https://i.stack.imgur.com/Prc4C.png" rel="nofollow noreferrer" style="margin: 0px; padding: 0px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 15px; vertical-align: baseline; --_li-fc:var(--theme-link-color); --_li-fc-hover:var(--theme-link-color-hover); --_li-fc-visited:var(--theme-link-color-visited); color: var(--theme-link-color); cursor: pointer; text-decoration: underline; user-select: auto; box-sizing: inherit;"><img src="https://i.stack.imgur.com/Prc4C.png" alt="enter image description here" style="margin: 0px; padding: 0px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 15px; vertical-align: bottom; box-sizing: inherit; max-width: 100%;"></a><br style="box-sizing: inherit;">I tried testing this by adding a pressure drop (called inlet in the immage) and changing it from very large to very low. (inlet_nominal in the code below) But what ever I do I never get points on the curve. Especially for low pressure drops I get values for flow that exceed 110m3/h (or 0.0235kg/s as calculated). For example when I set inlet_nominal = 20 Pa. The simulation gives a flow of 0.0258kg/s (*4680 = 120m3/h) and a pressure over the fan (fan.dp) of 29Pa. I would expect either a smaller pressure with that flow or a smaller flow with that pressure.</p><p style="margin: 0px 0px 1.1em; padding: 0px; border: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-alternates: inherit; font-weight: 400; font-stretch: inherit; line-height: inherit; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI Adjusted&quot;, &quot;Segoe UI&quot;, &quot;Liberation Sans&quot;, sans-serif; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 15px; vertical-align: baseline; box-sizing: inherit; clear: both; --_pr-img-mb:0; color: rgb(35, 38, 41); letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">The values for the pressure curve are in the table below. kg/s is calculated with a density of 1.3 kg/m3</p><div class="s-table-container" style="margin: 0px 0px 1.1em; padding: 0px; border: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-alternates: inherit; font-weight: 400; font-stretch: inherit; line-height: inherit; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI Adjusted&quot;, &quot;Segoe UI&quot;, &quot;Liberation Sans&quot;, sans-serif; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 15px; vertical-align: baseline; box-sizing: inherit; overflow-x: auto; color: rgb(35, 38, 41); letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

m3/h | kg/s | Pa
 -- | -- | --
 0 | 0 | 175
 10 | 0.00214 | 170*
 70 | 0.0150 | 100*
 105 | 0.0224 | 20*
 110 | 0.02354 | 0

</div><p style="margin: 0px 0px 1.1em; padding: 0px; border: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-alternates: inherit; font-weight: 400; font-stretch: inherit; line-height: inherit; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI Adjusted&quot;, &quot;Segoe UI&quot;, &quot;Liberation Sans&quot;, sans-serif; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 15px; vertical-align: baseline; box-sizing: inherit; clear: both; --_pr-img-mb:0; color: rgb(35, 38, 41); letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">The values marked * are in the per definition in the code example below where the fan curve is defined. (per(pressure(V_flow = {0.0002, 0.015, 0.0224},dp = {170, 100, 20})))<br style="box-sizing: inherit;">The other values I tried earlier. The results are similar.</p><pre style="margin: 0px 0px calc(1.5em); padding: var(--su12); border: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-alternates: inherit; font-weight: 400; font-stretch: inherit; line-height: var(--lh-md); font-family: var(--ff-mono); font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: var(--fs-body1); vertical-align: baseline; box-sizing: inherit; width: auto; max-height: 600px; overflow: auto; background-color: var(--highlight-bg); border-radius: var(--br-md); overflow-wrap: normal; color: var(--highlight-color); letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><code style="margin: 0px; padding: 0px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: var(--ff-mono); font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: var(--_pr-code-fs); vertical-align: baseline; box-sizing: inherit; background-color: transparent; white-space: inherit; color: var(--black-800); border-radius: 0px;">model FanTest01
  package Medium = Buildings.Media.Air;
  parameter Modelica.Units.SI.MassFlowRate m_flow_nominal = 0.0224 "Nominal mass flow rate";
  parameter Modelica.Units.SI.MassFlowRate m_flow_nominal_fan = 0.0128 "110m3/h Nominal mass flow rate for fan";
  parameter Modelica.Units.SI.PressureDifference dp_nominal = 5 "Nominal pressure difference";
  parameter Modelica.Units.SI.PressureDifference dp_nominal_fan = 117 "Nominal pressure difference for fan";
  parameter Modelica.Units.SI.PressureDifference inlet_nominal = 20 "Inlet pressure difference";
  parameter Modelica.Units.SI.Length steek = 0.44 "Horizontale steek tussen modulles";
  parameter Modelica.Units.SI.Length diameter = 0.08 "Buis diameter";
  // FAN
  Buildings.Fluid.Movers.SpeedControlled_y fan(redeclare package Medium = Medium,
  per(pressure(V_flow = {0.0002, 0.015, 0.0224},
    dp = {170, 100, 20})), 
    energyDynamics = Modelica.Fluid.Types.Dynamics.FixedInitial) "Fan" annotation(
    Placement(visible = true, transformation(origin = {70, 30}, extent = {{40, 40}, {60, 60}}, rotation = 0)));
  
  Buildings.Fluid.Sources.Boundary_pT sou(redeclare package Medium = Medium, T = 293.15, nPorts = 2, p = 101300, use_p_in = false) annotation(
    Placement(visible = true, transformation(origin = {82, 0}, extent = {{-82, 10}, {-62, 30}}, rotation = 0)));
  
  Buildings.Fluid.FixedResistances.HydraulicDiameter duct01(redeclare package Medium = Medium,
    dh = diameter, fac = 1, length = 0.44,
    m_flow_nominal = m_flow_nominal) annotation(
    Placement(visible = true, transformation(origin = {150, 80}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  
  Buildings.Fluid.FixedResistances.HydraulicDiameter duct02(redeclare package Medium = Medium,
    dh = diameter, fac = 1, length = 0.44,
    m_flow_nominal = m_flow_nominal) annotation(
    Placement(visible = true, transformation(origin = {90, 80}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  Buildings.Fluid.FixedResistances.PressureDrop inlet(redeclare package Medium = Medium,
    dp_nominal = inlet_nominal,
    m_flow_nominal = m_flow_nominal) annotation(
    Placement(visible = true, transformation(origin = {50, 50}, extent = {{-10, -10}, {10, 10}}, rotation = 90)));
  
  Modelica.Blocks.Sources.Ramp ramp(duration = 100, startTime = 10) annotation(
    Placement(visible = true, transformation(origin = {90, 110}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));

equation
  connect(fan.port_b, duct01.port_a) annotation(
    Line(points = {{130, 80}, {140, 80}}, color = {0, 127, 255}));
  connect(duct01.port_b, sou.ports[1]) annotation(
    Line(points = {{160, 80}, {170, 80}, {170, 20}, {20, 20}}, color = {0, 127, 255}));
  connect(ramp.y, fan.y) annotation(
    Line(points = {{101, 110}, {120, 110}, {120, 92}}, color = {0, 0, 127}));
  connect(fan.port_a, duct02.port_b) annotation(
    Line(points = {{110, 80}, {100, 80}}, color = {0, 127, 255}));
  connect(inlet.port_a, sou.ports[2]) annotation(
    Line(points = {{50, 40}, {50, 20}, {20, 20}}, color = {0, 127, 255}));
  connect(inlet.port_b, duct02.port_a) annotation(
    Line(points = {{50, 60}, {50, 80}, {80, 80}}, color = {0, 127, 255}));
  annotation(
    Diagram(coordinateSystem(preserveAspectRatio = true, extent = {{-200, 160}, {200, -240}})),
    Documentation(info = "&lt;html&gt;
&lt;p&gt;
Adapted from 'ClosedLoop_y' example.
&lt;/p&gt;
&lt;/html&gt;"),
    __Dymola_Commands(file = "modelica://Buildings/Resources/Scripts/Dymola/Fluid/Movers/Examples/ClosedLoop_y.mos" "Simulate and plot"),
    experiment(StopTime = 200, Tolerance = 1e-06, StartTime = 0, Interval = 1),
    uses(Buildings(version = "9.1.0"), Modelica(version = "4.0.0")),
    Icon(coordinateSystem(extent = {{-200, -260}, {200, 160}}, preserveAspectRatio = true)),
    version = "");
end FanTest01;
</code></pre><p style="margin: 0px; padding: 0px; border: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-alternates: inherit; font-weight: 400; font-stretch: inherit; line-height: inherit; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI Adjusted&quot;, &quot;Segoe UI&quot;, &quot;Liberation Sans&quot;, sans-serif; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 15px; vertical-align: baseline; box-sizing: inherit; clear: both; --_pr-img-mb:0; color: rgb(35, 38, 41); letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">What should I change to get simulation results closer to the desired fan curve from the supplier? It could well be I'm doing something silly wrong.</p>
</body>
</html>

**Comment by hcasperfu** [comment]
Greetings!
The fan curve can only be observed at constant speed. In your model, the speed signal to the fan is a ramp signal with varied speed.
In v9.1.0 there is one lower-level validation model `Buildings.Fluid.Movers.BaseClasses.Validation.FlowMachineInterface` from which you can look at the fan curve by plotting `eff.dp` vs `eff.V_flow`. However, this model validates the low-level class that handles the pressure curve. At the level of a full mover, things get more complicated.
You may refer to `Buildings.Fluid.Movers.Validation.NegativePressureOrFlow` to see how to force a pressure at constant speed to see the mover react to it with a certain flow rate, hence validating the pressure curve. But this model will be removed in the next release.
I will open an issue to add a validation model that can directly display the fan/pump curve, since this seems to be asked quite a few times recently.

**Reply by DynteqMS** [reply]
Thank you for your answer Hcasperfu,
I'm not sure I understand how this works completely.
In my understanding a fan curve shows how a fan will perform at a certain power (usually 100%). To find the flow and pressure you would plot the resistance curve (also flow vs pressure) of the connected system in the same graph. The two lines intersect at the equilibrium of pressure and flow.

The Ramp signal in my model is not important right now. For the first check I only look at the constant pressure after  t=100 where the input for the pump is 1. I run the simulation with different parameters. If I set the resistance very high then I get a high pressure and a small flow more or less as expected. But if I set the resistance low (small dp@inlet  and short ducts) the the flow and pressure are both above the curve. That is not the expected behavior.
It could be I'm doing something wrong in the simulation setup, or my understanding of how it works physically is incorrect, or it is a numeric quirk of how the thing is modeled.
Any hints are welcome.

**Reply by hcasperfu** [reply, marked as answer]
Greetings again!
The fan curve shows the pressure-flow relationship at a constant speed, instead of power. Other than that, I think your description of the physics is correct.
However, I think combining the fan curve and the system curve to validate the fan curve seems to bring in unnecessary complications for this purpose. A simple validation model that displays the pressure curve of the fan is currently being reviewed and should be available on the `master` branch soon (see #3376) and will go into the next release. The model will be `Buildings.Fluid.Movers.Validation.PressureCurve`.
Concerning the fan's operating point going above the curve, this is partly because the model does not restrict the speed of the pump to an upper limit, i.e. the Nrpm input is allowed to go above `speed_rpm_nominal` and therefore "above the curve".
Casper

**Reply by hcasperfu** [reply]
Greetings!
As a followup, there will be a new validation model `Buildings.Fluid.Movers.Validation.PressureCurve` that outputs the pressure curve from the mover model. Simply plot `mov.dpMachine` vs `mov.VMachine_flow`.
This is already available on the `master` branch (https://github.com/lbl-srg/modelica-buildings/commit/368f3a3bdd4d13e992626cf1e8e393b8ad26631a or later).

Cheers,
Casper

**Reply by DynteqMS** [reply]
Hi Casper,
Nice to hear the curve can be plotted in the near future, that will make validation a lot more intuitive.
A quick follow up question about the Nrpm or speed_rpm_nominal. I don't see these anywhere in the SpeedControlled_y pump. I assumed that the y input controlled either power or speed, and setting it to 1 would be nominal.
What does y control if not rpm?
Guess I missed the rpm part in SpeedControlled_Nrpm. Would SpeedControlled_Nrpm be better suited? And does this take the rpm as input? (and thereby prevent operation above the curve if set to nominal)
Cheers,
Martijn

**Reply by hcasperfu** [reply]
The `y` input is the normalised speed, i.e. _y = Nrpm / Nrpm_nominal_. Because internally the mover model converts the `Nrpm` signal (with an RPM unit) to a `y` signal (with a unit of "1") anyway, for the model with a _y_-input, any RPM information is redundant. The `_y` model always has _y_nominal = 1_.
Please note that the `_Nrpm` model will be moved to the `Obsolete` package in the upcoming release. This is to simplify the implementation and for its lack of use (see https://github.com/ibpsa/modelica-ibpsa/issues/1704). In the linked post we also briefly discussed whether we want to put limits to the input signal, so that `y` is limited between 0 and 1, as you suggested. This was rejected because (a) this, again, would complicate the implementation and (b) such a limiter is better positioned in the control sequence outside of the component model.
If it is more convenient for you to use a RPM input, it can be scaled to a y signal by a gain block, i.e. `Modelica.Blocks.Math.Gain` or `Buildings.Controls.OBC.CDL.Continuous.MultiplyByParameter`. To limit the control signal within 0 to 1, use `Buildings.Controls.OBC.CDL.Continuous.Limiter`.

**Reply by DynteqMS** [reply]
Hi Casper,
Thanks for your time explaining. Sorry but I'm still a bit lost. In my model I have a ramp on the input of y. that rises to 1 and then stays there. So I don't understand how the fan can perform above the curve.

Seeing the discussion in the 1704 thread you linked, I don't think switching to Nrpm would help me.

**Reply by hcasperfu** [reply]
Can you please provide details how you parameterrised the model where you observed this issue?
As I run the code you provided in the original post, after the modification of `fan.use_inputFilter=false`, it looks fine. Except at the beginning of the simulation when the system was not yet stabalised, the pressure and flow behaved as expected and finally landed exactly on the curve.

<img width="958" alt="Screenshot 2023-05-09 at 13 55 35" src="https://github.com/lbl-srg/modelica-buildings/assets/91219266/ae704f62-88ae-4868-b72c-958ada6de507">

**Reply by DynteqMS** [reply]
Hi Casper,
Thanks for your perseverance and patience. 
When I set inlet_nominal to 20 Pa. The simulation gives a flow of 0.0258kg/s (*4680 = 120m3/h) and a pressure over the fan (fan.dp) of 29Pa. I would expect either a smaller pressure with that flow or a smaller flow with that pressure. This was still true this morning when I re-ran the simulation.
Maybe the fan.use_inputFilter=false parameter does the trick, what does this do? In the GUI it says: "= true, if speed is filtered with a 2nd order CriticalDamping filter" but I have no idea what that means.

Since I touched the input filter setting the test no longer simulates. I'm not sure what changed.
warnings/errors:
```
[1] 10:08:17 Scripting Notification
Automatically loaded package Complex 4.0.0 due to uses annotation from Modelica.

[2] 10:08:17 Scripting Notification
Automatically loaded package ModelicaServices 4.0.0 due to uses annotation from Modelica.

[3] 10:08:17 Scripting Notification
Skip loading Modelica version default since latest version is already loaded because of the setting Load latest Modelica version on startup.

[4] 10:08:33 Scripting Notification
Automatically loaded package Buildings 9.1.0 due to uses annotation from FanTest01.

[1] 10:13:23 Translation Error
[C:/OM120/OM64bit/OMCompiler/Compiler/NFFrontEnd/NFCeval.mo: 1088:9-1088:67]: Internal error NFCeval.evalBinaryMul failed to evaluate ‘0.0 / 0.0 * 0.0025‘

```
Using OMEdit version:
```
Connected to OpenModelica v1.20.0 (64-bit)
Connected to OMSimulator v2.1.1.post188-gaf996ad-mingw
```

Code:
```
model FanTest01
  package Medium = Buildings.Media.Air;
  parameter Modelica.Units.SI.MassFlowRate m_flow_nominal = 0.0224 "Nominal mass flow rate";
  parameter Modelica.Units.SI.MassFlowRate m_flow_nominal_fan = 0.0128 "110m3/h Nominal mass flow rate for fan";
  parameter Modelica.Units.SI.PressureDifference dp_nominal = 5 "Nominal pressure difference";
  parameter Modelica.Units.SI.PressureDifference dp_nominal_fan = 117 "Nominal pressure difference for fan";
  parameter Modelica.Units.SI.PressureDifference inlet_nominal = 20 "Inlet pressure difference";
  parameter Modelica.Units.SI.Length steek = 0.44 "Horizontale steek tussen modulles";
  parameter Modelica.Units.SI.Length diameter = 0.08 "Buis diameter";
  // FAN
  Buildings.Fluid.Movers.SpeedControlled_y fan(redeclare package Medium = Medium, 
    energyDynamics = Modelica.Fluid.Types.Dynamics.FixedInitial, use_inputFilter = true) "Fan" annotation(
    Placement(visible = true, transformation(origin = {70, 30}, extent = {{40, 40}, {60, 60}}, rotation = 0)));
  
  Buildings.Fluid.Sources.Boundary_pT sou(redeclare package Medium = Medium, T = 293.15, nPorts = 2, p = 101300, use_p_in = false) annotation(
    Placement(visible = true, transformation(origin = {82, 0}, extent = {{-82, 10}, {-62, 30}}, rotation = 0)));
  
  Buildings.Fluid.FixedResistances.HydraulicDiameter duct01(redeclare package Medium = Medium,
    dh = diameter, fac = 1, length = 0.44,
    m_flow_nominal = m_flow_nominal) annotation(
    Placement(visible = true, transformation(origin = {150, 80}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  
  Buildings.Fluid.FixedResistances.HydraulicDiameter duct02(redeclare package Medium = Medium,
    dh = diameter, fac = 1, length = 0.44,
    m_flow_nominal = m_flow_nominal) annotation(
    Placement(visible = true, transformation(origin = {90, 80}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  Buildings.Fluid.FixedResistances.PressureDrop inlet(redeclare package Medium = Medium,
    dp_nominal = inlet_nominal,
    m_flow_nominal = m_flow_nominal) annotation(
    Placement(visible = true, transformation(origin = {50, 50}, extent = {{-10, -10}, {10, 10}}, rotation = 90)));
  
  Modelica.Blocks.Sources.Ramp ramp(duration = 100, startTime = 10) annotation(
    Placement(visible = true, transformation(origin = {90, 110}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));

equation
  connect(fan.port_b, duct01.port_a) annotation(
    Line(points = {{130, 80}, {140, 80}}, color = {0, 127, 255}));
  connect(duct01.port_b, sou.ports[1]) annotation(
    Line(points = {{160, 80}, {170, 80}, {170, 20}, {20, 20}}, color = {0, 127, 255}));
  connect(ramp.y, fan.y) annotation(
    Line(points = {{101, 110}, {120, 110}, {120, 92}}, color = {0, 0, 127}));
  connect(fan.port_a, duct02.port_b) annotation(
    Line(points = {{110, 80}, {100, 80}}, color = {0, 127, 255}));
  connect(inlet.port_a, sou.ports[2]) annotation(
    Line(points = {{50, 40}, {50, 20}, {20, 20}}, color = {0, 127, 255}));
  connect(inlet.port_b, duct02.port_a) annotation(
    Line(points = {{50, 60}, {50, 80}, {80, 80}}, color = {0, 127, 255}));
  annotation(
    Diagram(coordinateSystem(preserveAspectRatio = true, extent = {{-200, 160}, {200, -240}})),
    Documentation(info = "<html>
<p>
Adapted from 'ClosedLoop_y' example.
</p>
</html>"),
    __Dymola_Commands(file = "modelica://Buildings/Resources/Scripts/Dymola/Fluid/Movers/Examples/ClosedLoop_y.mos" "Simulate and plot"),
    experiment(StopTime = 200, Tolerance = 1e-06, StartTime = 0, Interval = 1),
    uses(Buildings(version = "9.1.0"), Modelica(version = "4.0.0")),
    Icon(coordinateSystem(extent = {{-200, -260}, {200, 160}}, preserveAspectRatio = true)),
    version = "");
end FanTest01;
```

**Reply by hcasperfu** [reply]
Greetings!

In the new code you pasted, the fan curve `fan.per` is no longer specified. This leaves the fan curve with zeros (default values) which then leads the simulator to attempt evaluating 0/0 and throw out an error. Adding `fan.per` back fixes the model.

The signal filter smoothens the input signal to the fan which helps the solver. See the figure below and the "Start-up and shut-down transients" section in `usersguide`. I set it to false to remove these transient effects and try to see the immediate response of the fan (i.e. the actual speed is always equal to the input signal without smoothening).
<img width="414" alt="Screenshot 2023-05-11 at 08 50 38" src="https://github.com/lbl-srg/modelica-buildings/assets/91219266/2ec630c7-709d-42de-87c3-8c33b3b648e9">

I'm not sure your unit conversion is correct. The volumetric flow rate is directly available in the results as `VMachine_flow`. It's value is inline with the values in the pressure curve.
Also note that the flow rate of the fan curve is the **volumetric** flow rate instead of mass. Looking back at the table you have in the original post you seem to have the mass unit for them.
<img width="294" alt="Screenshot 2023-05-11 at 09 03 00" src="https://github.com/lbl-srg/modelica-buildings/assets/91219266/ff025372-68f1-4eba-aaa3-a1d013f429f9">

**Reply by DynteqMS** [reply]
Hi Casper,
Thanks again for your time.
I'm ashamed to admit there was an error in my unit conversion like you said. I must have missed that part of your comment the first time. 
The "per" definition was indeed missing in my code. I am happy to report it is working again, and gives the expected results now.
I also found the cause for the missing "per" definition. If I change the "fan.use_inputFilter=false" through the GUI it removes the "per" definition in my code. I think this may be a OMEdit or Openmodelica burg. Do you know where I should report that?

Kind regards,
Martijn

**Reply by hcasperfu** [reply]
Greetings.
I'm glad to hear that the model is running for you.
You can report this bug by opening an issue at https://github.com/OpenModelica/OpenModelica. A bug report template is integrated on their github site which will guide you through the process.

I was able to reproduce this bug with OMEdit on my machine as well.
My versions:
**OpenModelica 1.21.0
OMSimulator 2.1.1~188-gaf996ad-linux
OS: Ubuntu 64-bit 20.04.3**

You can tag me in the post and provide this additional piece of information if you want.

Cheers,
Casper
